--- original.py	2025-12-03 21:01:21.371655371 +0200
+++ patched.py	2025-12-03 21:01:26.371447811 +0200
@@ -173,6 +173,8 @@
             ) from e
 
     gpu_memory = get_vram(device)
+    if gpu_memory >= 22: 
+        batch_ratio = 32
     if gpu_memory >= 16:
         batch_ratio = 16
     elif gpu_memory >= 12:
@@ -188,14 +190,15 @@
             f'You can set MINERU_VIRTUAL_VRAM_SIZE environment variable to adjust GPU memory allocation.'
     )
 
-    # 检测torch的版本号
-    import torch
-    from packaging import version
-    if version.parse(torch.__version__) >= version.parse("2.8.0") or str(device).startswith('mps'):
-        enable_ocr_det_batch = False
-    else:
-        enable_ocr_det_batch = True
-
+    # # 检测torch的版本号
+    # import torch
+    # from packaging import version
+    # if version.parse(torch.__version__) >= version.parse("2.8.0") or str(device).startswith('mps'):
+    #     enable_ocr_det_batch = False
+    # else:
+    #     enable_ocr_det_batch = True
+    enable_ocr_det_batch = True
+    print(f"enable_ocr_det_batch: {enable_ocr_det_batch}")
     batch_model = BatchAnalyze(model_manager, batch_ratio, formula_enable, table_enable, enable_ocr_det_batch)
     results = batch_model(images_with_extra_info)
 
